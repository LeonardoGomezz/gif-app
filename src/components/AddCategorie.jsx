/* eslint-disable react/prop-types */
import { useState } from "react"

export const AddCategorie = ({onNewCategorie}) => {

  const [inputValue, setInputValue] = useState('')

  const onInputChange = ({target}) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) =>{
    event.preventDefault()
    if(inputValue.trim().length <= 1 )return
    onNewCategorie( inputValue.trim() )
    //setCategories(categories => [inputValue, ...categories])
    setInputValue('')
  }

  return (
    <form onSubmit={onSubmit}>
    <input 
    type="text" 
    placeholder="Buscar gifs" 
    value={inputValue}
    onChange={ onInputChange }
    />
    </form>
  )
}
