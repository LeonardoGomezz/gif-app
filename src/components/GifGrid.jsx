import PropTypes from 'prop-types';
import { getGitf } from '../helpers/getGifs';
import { useEffect, useState } from 'react';
import { GiftItem } from './GiftItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([])

  const getImages = async() =>{
    const newImages = await getGitf(category)
    setImages(newImages)
  }

  useEffect( () => {
    getImages()
  }, [])

  return (
    <>
    <h3>{category}</h3>
      <div className='card-grid'>

      {
        images.map((image) =>(
          <GiftItem key={image.id}
          {...image}
          />
          ))
        }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
