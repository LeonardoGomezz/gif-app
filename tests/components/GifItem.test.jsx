import { render, screen } from "@testing-library/react"
import { GiftItem } from "../../src/components/GiftItem"

describe('Pruebas en <GifItem/>', () =>{
  const title = 'saitama'
  const url = 'https://cdn.pixabay.com/photo/2023/08/24/19/58/saitama-8211499_960_720.png'

  test('debe hacer march con el snap shot', () => { 
   const {container} = render( <GiftItem title={title} url={url}/>)
   expect(container).toMatchSnapshot()
  })

  test('debe mostrar la imagen y el alt indicado', () => { 
    render( <GiftItem title={title} url={url}/>)
    // screen.debug()
    // expect((screen.getByRole('img').src)).toBe( url )
    const { src, alt } = screen.getByRole('img')
    expect( src ).toBe( url )
    expect( alt ).toBe( alt )
   })

   test('Debe mostrar el titulo del componente', () => { 
    render( <GiftItem title={title} url={url}/>)
    expect( screen.getByText( title ) ).toBeTruthy()
    })
})