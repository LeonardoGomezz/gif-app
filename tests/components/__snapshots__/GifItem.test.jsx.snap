// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe hacer march con el snap shot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://cdn.pixabay.com/photo/2023/08/24/19/58/saitama-8211499_960_720.png"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
